volue_insight_timeseries/curves.py

```diff
@@ -148,6 +150,62 @@ def get_data(self, data_from=None, data_to=None, time_zone=None, filter=None,
             return result
         return util.TS(input_dict=result, curve_type=util.TIME_SERIES)
 
+    def get_data_range(self, date_from=None, date_to=None, output_time_zone=None) -> Optional[util.Range]:
+        """ Retrieves a range of data from a Time Series curve
+
+        This method fetches a range of data from the curve between two timestamps. It's mainly used for determining
```

Comment:
Customers have access to a given range of the curve, which is not always the full range (https://volueinsight.com/docs/curve-types.html#access-range). For example: customer AAA has paid for access to curve C1 from 2020 till 2022. C1 has data from 2010 - 2024. If AAA queries the range of C1, do we return 2020-2022 or 2010-2024?

Reply:
With @hieuhc help i found that this should return 2010-2024. In order to check it  as Hieu put it: _"we would need to change the restriction removing internal roles for this api, and deploy to prod. don’t think you can test at this stage."_ Looking through wapi source code there was no clipping by access. What should be the expected behavior? I would assume it should be clipped on the wapi side so that there is no leakage on the amount of data that we have?

Reply:
@stigalvestad I have updated the docs to reflect the fact, that the get data range does include all data even in case the client does not have access to it.